java-checks/src/test/java/org/sonar/java/checks/security/AndroidPersistentUniqueIdentifierCheckTest.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+package org.sonar.java.checks.security;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.java.checks.verifier.CheckVerifier;
+
+import static org.sonar.java.checks.verifier.TestUtils.mainCodeSourcesPath;
+
+class AndroidPersistentUniqueIdentifierCheckTest {
+
+  @Test
+  void test() {
+    CheckVerifier.newVerifier()
+      .onFile(mainCodeSourcesPath("checks/security/AndroidPersistentUniqueIdentifierCheckSample.java"))
+      .withCheck(new AndroidPersistentUniqueIdentifierCheck())
+      .verifyIssues();
+  }
+
```

Comment:
We could probably add a test here without semantic to detect regression or changes in the way the check operates
```suggestion
@Test
  void without_semantic() {
    CheckVerifier.newVerifier()
      .onFile(mainCodeSourcesPath("checks/security/AndroidPersistentUniqueIdentifierCheckSample.java"))
      .withCheck(new AndroidPersistentUniqueIdentifierCheck())
      .withoutSemantic()
      .verifyNoIssues();
  }
```